readme.md

```diff
@@ -146,3 +148,11 @@ things before they become added to the project:
 3. the feature has been discussed in the issue list beforehand
 
 We automate all of our testing and use pre-commit hooks to keep the code working.
+
+### Implementing a new preprocessor
+
+When creating a completely new transformer in `sklego.preprocessing`, make sure to implement `get_feature_names_out` functionality. 
+
+The preferred method is to add [_ClassNamePrefixFeaturesOutMixin](https://github.com/scikit-learn/scikit-learn/blob/626b4608d4f840af7c37bff2ccb38fcfd2ef594f/sklearn/base.py#L868) as a base class\
+and make sure `self._n_features_out` (i.e. the number of output features) is defined in the `fit` method of the new preprocessor.
+
```

Comment:
@MBrouns I'm wondering. Would it be better to have folks import a hidden mixin, or just to have them implement `get_feature_names_out` themselves? Part of me is uneasy about relying on a non-public part of the sklearn api.

Reply:
@CarloLepelaars also, seen [this](https://scikit-lego.netlify.app/contribution.html)?

Reply:
interesting that this is part of the private API. Maybe copy the sklearn implementation into one of our own files then and use that instead for the time being?

Reply:
Fair point, the Mixin seems like the most clean implementation, but agree it is not a good idea to use Mixins from the non-public API. Would be interesting to ask an sklearn core maintainer about this decision. Not sure if they are aware that there are open-source projects like scikit-lego that are tightly built on sklearn. Seems like a good argument to make these kinds of Mixins part of the public API.

@MBrouns, sounds like a good intermediate step. Will put it in `common.py` with the `TrainOnlyTransformerMixin`.

Reply:
> That already looks great! Perhaps we can add a link to here in the "New Features" section of the readme?

Sounds good.

Reply:
Mhm, yeah, this is tricky territory.

If their internal class has many more links to internal objects it'll be even riskier to rely on it. 

Open to ideas though, there doesn't seem to be an "obvious" solution here.